Fix `TestArgoCDWaitEnsureAppIsNotCrashing`

```
=== RUN   TestArgoCDWaitEnsureAppIsNotCrashing
time="2019-08-02T13:46:13-07:00" level=info msg="kubectl delete ns -l e2e.argoproj.io=true --field-selector status.phase=Active --wait=false" dir= execID=qKxbK
time="2019-08-02T13:46:13-07:00" level=debug msg="namespace \"argocd-e2e-test-comparison-fails-if-cluster-not-added-ngasp\" deleted\n" duration=90.281843ms execID=qKxbK
time="2019-08-02T13:46:13-07:00" level=info msg="Starting configmap/secret informers"
time="2019-08-02T13:46:13-07:00" level=info msg="configmap informer cancelled"
time="2019-08-02T13:46:13-07:00" level=info msg="secrets informer cancelled"
time="2019-08-02T13:46:13-07:00" level=info msg="Configmap/secret informer synced"
time="2019-08-02T13:46:16-07:00" level=info msg="mkdir -p /tmp/argo-e2e" dir= execID=dvFl0
time="2019-08-02T13:46:16-07:00" level=debug duration=3.813886ms execID=dvFl0
time="2019-08-02T13:46:16-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/tls" dir= execID=sl2F4
time="2019-08-02T13:46:16-07:00" level=debug duration=3.199547ms execID=sl2F4
time="2019-08-02T13:46:16-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/ssh" dir= execID=WbTox
time="2019-08-02T13:46:16-07:00" level=debug duration=2.93566ms execID=WbTox
time="2019-08-02T13:46:16-07:00" level=info msg="cp -Rf testdata /tmp/argo-e2e/testdata.git" dir= execID=n3tlk
time="2019-08-02T13:46:16-07:00" level=debug duration=44.633858ms execID=n3tlk
time="2019-08-02T13:46:16-07:00" level=info msg="chmod 777 ." dir=/tmp/argo-e2e/testdata.git execID=vgMRU
time="2019-08-02T13:46:16-07:00" level=debug duration=3.527035ms execID=vgMRU
time="2019-08-02T13:46:16-07:00" level=info msg="git init" dir=/tmp/argo-e2e/testdata.git execID=ZIDNF
time="2019-08-02T13:46:16-07:00" level=debug msg="Initialized empty Git repository in /private/tmp/argo-e2e/testdata.git/.git/\n" duration=14.063986ms execID=ZIDNF
time="2019-08-02T13:46:16-07:00" level=info msg="git add ." dir=/tmp/argo-e2e/testdata.git execID=kGSPt
time="2019-08-02T13:46:16-07:00" level=debug duration=54.487648ms execID=kGSPt
time="2019-08-02T13:46:16-07:00" level=info msg="git commit -q -m initial commit" dir=/tmp/argo-e2e/testdata.git execID=fnFlK
time="2019-08-02T13:46:16-07:00" level=debug duration=62.144632ms execID=fnFlK
time="2019-08-02T13:46:16-07:00" level=info msg="kubectl create ns argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj" dir= execID=5TxKK
time="2019-08-02T13:46:16-07:00" level=debug msg="namespace/argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj created\n" duration=99.92551ms execID=5TxKK
time="2019-08-02T13:46:16-07:00" level=info msg="kubectl label ns argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj e2e.argoproj.io=true" dir= execID=lJ0dP
time="2019-08-02T13:46:16-07:00" level=debug msg="namespace/argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj labeled\n" duration=137.198123ms execID=lJ0dP
time="2019-08-02T13:46:16-07:00" level=info msg="clean state" duration=3.056766013s id=test-argo-cd-wait-ensure-app-is-not-crashing-itudj name=test-argo-cd-wait-ensure-app-is-not-crashing
time="2019-08-02T13:46:16-07:00" level=info msg="../../dist/argocd app create test-argo-cd-wait-ensure-app-is-not-crashing --repo file:///tmp/argo-e2e/testdata.git --path guestbook --dest-server https://kubernetes.default.svc --dest-namespace argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj --project default --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=MDZWA
time="2019-08-02T13:46:17-07:00" level=debug msg="application 'test-argo-cd-wait-ensure-app-is-not-crashing' created\n" duration=466.213024ms execID=MDZWA
time="2019-08-02T13:46:17-07:00" level=info msg="polling for expectation" message="found success with message 'application 'test-argo-cd-wait-ensure-app-is-not-crashing' created'" state=succeeded
time="2019-08-02T13:46:17-07:00" level=info msg="../../dist/argocd app sync test-argo-cd-wait-ensure-app-is-not-crashing --timeout 5 --prune --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=4r11c
time="2019-08-02T13:46:18-07:00" level=debug msg="TIMESTAMP  GROUP        KIND   NAMESPACE                  NAME    STATUS   HEALTH        HOOK  MESSAGE\n2019-08-02T13:46:17-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui  OutOfSync  Missing              \n2019-08-02T13:46:17-07:00   apps  Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui  OutOfSync  Missing              \n2019-08-02T13:46:17-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n\nName:               test-argo-cd-wait-ensure-app-is-not-crashing\nProject:            default\nServer:             https://kubernetes.default.svc\nNamespace:          argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj\nURL:                http://localhost:8080/applications/test-argo-cd-wait-ensure-app-is-not-crashing\nRepo:               file:///tmp/argo-e2e/testdata.git\nTarget:             \nPath:               guestbook\nSync Policy:        <none>\nSync Status:        Synced to  (847d627)\nHealth Status:      Healthy\n\nOperation:          Sync\nSync Revision:      847d627810076e07b9ef3e6e38abc079a966e6f6\nPhase:              Succeeded\nStart:              2019-08-02 13:46:17 -0700 PDT\nFinished:           2019-08-02 13:46:18 -0700 PDT\nDuration:           1s\nMessage:            successfully synced (all tasks run)\n\nGROUP  KIND        NAMESPACE                                                      NAME          STATUS  HEALTH   HOOK  MESSAGE\n       Service     argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        service/guestbook-ui created\napps   Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        deployment.apps/guestbook-ui created\n" duration=1.195229539s execID=4r11c
time="2019-08-02T13:46:18-07:00" level=info msg="polling for expectation" message="found success with message 'TIMESTAMP  GROUP        KIND   NAMESPACE                  NAME    STATUS   HEALTH        HOOK  MESSAGE\n2019-08-02T13:46:17-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui  OutOfSync  Missing              \n2019-08-02T13:46:17-07:00   apps  Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui  OutOfSync  Missing              \n2019-08-02T13:46:17-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n\nName:               test-argo-cd-wait-ensure-app-is-not-crashing\nProject:            default\nServer:             https://kubernetes.default.svc\nNamespace:          argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj\nURL:                http://localhost:8080/applications/test-argo-cd-wait-ensure-app-is-not-crashing\nRepo:               file:///tmp/argo-e2e/testdata.git\nTarget:             \nPath:               guestbook\nSync Policy:        <none>\nSync Status:        Synced to  (847d627)\nHealth Status:      Healthy\n\nOperation:          Sync\nSync Revision:      847d627810076e07b9ef3e6e38abc079a966e6f6\nPhase:              Succeeded\nStart:              2019-08-02 13:46:17 -0700 PDT\nFinished:           2019-08-02 13:46:18 -0700 PDT\nDuration:           1s\nMessage:            successfully synced (all tasks run)\n\nGROUP  KIND        NAMESPACE                                                      NAME          STATUS  HEALTH   HOOK  MESSAGE\n       Service     argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        service/guestbook-ui created\napps   Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        deployment.apps/guestbook-ui created'" state=succeeded
time="2019-08-02T13:46:18-07:00" level=info msg="polling for expectation" message="sync status to be Synced, is Synced" state=succeeded
time="2019-08-02T13:46:18-07:00" level=info msg="polling for expectation" message="health to should Healthy, is Healthy" state=succeeded
time="2019-08-02T13:46:18-07:00" level=info msg="../../dist/argocd app set test-argo-cd-wait-ensure-app-is-not-crashing --path crashing-guestbook --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=2Dfrl
time="2019-08-02T13:46:18-07:00" level=debug duration=346.540515ms execID=2Dfrl
time="2019-08-02T13:46:18-07:00" level=error msg="`../../dist/argocd app set test-argo-cd-wait-ensure-app-is-not-crashing --path crashing-guestbook --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure` failed exit status 1: time=\"2019-08-02T13:46:18-07:00\" level=fatal msg=\"rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: Unable to generate manifests in crashing-guestbook: rpc error: code = Unknown desc = `kustomize build /var/folders/v5/0lllbd_s0ldg1dr4qxm7vw0scm2ffd/T/file:___tmp_argo-e2e_testdata/crashing-guestbook` failed exit status 1: Error: json: cannot unmarshal string into Go struct field Kustomization.patches of type types.Patch\"" execID=2Dfrl
time="2019-08-02T13:46:18-07:00" level=info msg="../../dist/argocd app sync test-argo-cd-wait-ensure-app-is-not-crashing --timeout 5 --prune --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=aelxh
time="2019-08-02T13:46:19-07:00" level=debug msg="TIMESTAMP                  GROUP        KIND   NAMESPACE                                                                     NAME    STATUS   HEALTH        HOOK  MESSAGE\n2019-08-02T13:46:18-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n2019-08-02T13:46:18-07:00   apps  Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n\nName:               test-argo-cd-wait-ensure-app-is-not-crashing\nProject:            default\nServer:             https://kubernetes.default.svc\nNamespace:          argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj\nURL:                http://localhost:8080/applications/test-argo-cd-wait-ensure-app-is-not-crashing\nRepo:               file:///tmp/argo-e2e/testdata.git\nTarget:             \nPath:               guestbook\nSync Policy:        <none>\nSync Status:        Synced to  (847d627)\nHealth Status:      Healthy\n\nOperation:          Sync\nSync Revision:      847d627810076e07b9ef3e6e38abc079a966e6f6\nPhase:              Succeeded\nStart:              2019-08-02 13:46:18 -0700 PDT\nFinished:           2019-08-02 13:46:19 -0700 PDT\nDuration:           1s\nMessage:            successfully synced (all tasks run)\n\nGROUP  KIND        NAMESPACE                                                      NAME          STATUS  HEALTH   HOOK  MESSAGE\n       Service     argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        service/guestbook-ui unchanged\napps   Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        deployment.apps/guestbook-ui unchanged\n" duration=1.187713698s execID=aelxh
time="2019-08-02T13:46:19-07:00" level=info msg="polling for expectation" message="found success with message 'TIMESTAMP                  GROUP        KIND   NAMESPACE                                                                     NAME    STATUS   HEALTH        HOOK  MESSAGE\n2019-08-02T13:46:18-07:00            Service  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n2019-08-02T13:46:18-07:00   apps  Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj          guestbook-ui    Synced  Healthy              \n\nName:               test-argo-cd-wait-ensure-app-is-not-crashing\nProject:            default\nServer:             https://kubernetes.default.svc\nNamespace:          argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj\nURL:                http://localhost:8080/applications/test-argo-cd-wait-ensure-app-is-not-crashing\nRepo:               file:///tmp/argo-e2e/testdata.git\nTarget:             \nPath:               guestbook\nSync Policy:        <none>\nSync Status:        Synced to  (847d627)\nHealth Status:      Healthy\n\nOperation:          Sync\nSync Revision:      847d627810076e07b9ef3e6e38abc079a966e6f6\nPhase:              Succeeded\nStart:              2019-08-02 13:46:18 -0700 PDT\nFinished:           2019-08-02 13:46:19 -0700 PDT\nDuration:           1s\nMessage:            successfully synced (all tasks run)\n\nGROUP  KIND        NAMESPACE                                                      NAME          STATUS  HEALTH   HOOK  MESSAGE\n       Service     argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        service/guestbook-ui unchanged\napps   Deployment  argocd-e2e-test-argo-cd-wait-ensure-app-is-not-crashing-itudj  guestbook-ui  Synced  Healthy        deployment.apps/guestbook-ui unchanged'" state=succeeded
time="2019-08-02T13:46:19-07:00" level=info msg="polling for expectation" message="sync status to be Synced, is Synced" state=succeeded
time="2019-08-02T13:46:19-07:00" level=info msg="polling for expectation" message="health to should Degraded, is Healthy" state=pending
time="2019-08-02T13:46:22-07:00" level=info msg="polling for expectation" message="health to should Degraded, is Healthy" state=pending
time="2019-08-02T13:46:25-07:00" level=info msg="polling for expectation" message="health to should Degraded, is Healthy" state=pending
time="2019-08-02T13:46:28-07:00" level=info msg="polling for expectation" message="health to should Degraded, is Healthy" state=pending
time="2019-08-02T13:46:31-07:00" level=info msg="polling for expectation" message="health to should Degraded, is Healthy" state=pending
--- FAIL: TestArgoCDWaitEnsureAppIsNotCrashing (21.31s)
    app_management_test.go:178: 
        	Error Trace:	app_management_test.go:178
        	            				consequences.go:44
        	            				app_management_test.go:176
        	Error:      	Received unexpected error:
        	            	`../../dist/argocd app set test-argo-cd-wait-ensure-app-is-not-crashing --path crashing-guestbook --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure` failed exit status 1: time="2019-08-02T13:46:18-07:00" level=fatal msg="rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: Unable to generate manifests in crashing-guestbook: rpc error: code = Unknown desc = `kustomize build /var/folders/v5/0lllbd_s0ldg1dr4qxm7vw0scm2ffd/T/file:___tmp_argo-e2e_testdata/crashing-guestbook` failed exit status 1: Error: json: cannot unmarshal string into Go struct field Kustomization.patches of type types.Patch"
        	Test:       	TestArgoCDWaitEnsureAppIsNotCrashing
    app_management_test.go:184: timeout waiting, health to should Degraded, is Healthy
```

--- a/test/e2e/app_management_test.go
+++ b/test/e2e/app_management_test.go
@@ -164,24 +164,15 @@ func TestComparisonFailsIfClusterNotAdded(t *testing.T) {
 		Expect(DoesNotExist())
 }
 
-func TestArgoCDWaitEnsureAppIsNotCrashing(t *testing.T) {
+func TestCannotSetInvalidPath(t *testing.T) {
 	Given(t).
 		Path(guestbookPath).
 		When().
 		Create().
-		Sync().
+		IgnoreErrors().
+		AppSet("--path", "garbage").
 		Then().
-		Expect(SyncStatusIs(SyncStatusCodeSynced)).
-		Expect(HealthIs(HealthStatusHealthy)).
-		And(func(app *Application) {
-			_, err := fixture.RunCli("app", "set", app.Name, "--path", "crashing-guestbook")
-			assert.NoError(t, err)
-		}).
-		When().
-		Sync().
-		Then().
-		Expect(SyncStatusIs(SyncStatusCodeSynced)).
-		Expect(HealthIs(HealthStatusDegraded))
+		Expect(Error("", "app path does not exist"))
 }
 
 func TestManipulateApplicationResources(t *testing.T) {

